### PR TITLE
Example: Making fine-tuning LFM2.5-VL-1.6B work for datasets with system prompt and vision layer fine-tuning.

### PR DESCRIPTION
**Pull request by urbanspr1nter**
See: https://github.com/unslothai/unsloth/issues/3938

Currently, `unsloth` seems to be breaking when trying to fine tune `LiquidAI/LFM2.5-VL-1.6B`. 

Reasons:
1. Chat template for the model assumes system prompt is a string, not a list. Normalization in vision utils will assume array and will break the collation process.
2. Vision layers cannot be fine tuned because the encoder is SigLIP2.

### CODE REVIEW

File: unsloth_zoo/vision_utils.py [hunk 1]
```diff
@@ -891,6 +894,11 @@ def _validate_and_normalize_first_message(self, messages):
         content = message.get("content")
         if isinstance(content, str):
             message["content"] = [{"type": "text", "text": content}]
+        if isinstance(content, list) and \
```
[the remark this answers is not in the document]
Comment (urbanspr1nter) [reply unsloth_zoo/vision_utils.py]: Hahaha thanks. Bad copy and paste

**Comment by danielhanchen** [comment]
Confirmed the collator path in unsloth_zoo/vision_utils.py still normalizes the first message to a list, which breaks the LFM2-VL template, so that part is worth fixing. The peft_utils hook was rewritten on main and no longer has the config-name branch, so could you drop that hunk and avoid the top level import of Lfm2VlProcessor since it will fail on transformers builds without that class?

**Comment by danielhanchen** [comment]
The system prompt normalization gap is still there in unsloth_zoo/vision_utils.py::_validate_and_normalize_first_message, but requires_grad_pre_hook in peft_utils.py was rewritten since you opened this, so that hunk no longer applies and needs a rebase. Could you also move from transformers import Lfm2VlProcessor off module scope and resolve it lazily, since older transformers versions without that class would fail to import vision_utils at all?

**Comment by danielhanchen** [comment]
Confirmed the system prompt still gets normalized to a list in vision_utils.py `_validate_and_normalize_first_message`, so that part is real, but importing Lfm2VlProcessor at module scope will break imports on transformers versions without it, so could you make it a lazy or guarded lookup? The `requires_grad_pre_hook` half no longer applies since that vision-config branch was replaced on main, so please rebase and drop it.
